### PR TITLE
upgrade react-script to v4

### PR DESCRIPTION
**Pull request by Nexucis**
Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Comment by Nexucis** [comment]
should make the PR #9284 a bit lighter and should help to debug it.

File: web/ui/react-app/package.json [hunk 1]
```diff
@@ -37,20 +36,18 @@
...
     "tempusdominus-bootstrap-4": "^5.1.2",
     "tempusdominus-core": "^5.0.3",
-    "typescript": "^3.3.3",
     "use-media": "^1.4.0"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --runInBand",
+    "test": "react-scripts test --runInBand --resetMocks=false",
```
Comment (Nexucis) [inline web/ui/react-app/package.json:50]: this weird thing is coming from this issue : https://github.com/facebook/create-react-app/issues/10571#issuecomment-784923197

**Comment by Nexucis** [comment]
ah yeah I remembered that part. Warnings are no treated as error in a CI.
It was the same in https://github.com/prometheus/prometheus/pull/8742 and I think we decided to solve all warning at the end.

File: web/ui/react-app/src/pages/graph/ExpressionInput.tsx [hunk 2]
```diff
@@ -37,64 +37,69 @@ class ExpressionInput extends Component<ExpressionInputProps, ExpressionInputSta
...
-            const matches = this.getSearchMatches(inputValue!, items);
+            const matches = this.getSearchMatches(inputValue, items);
             return !matches.length
               ? acc
               : [
                   ...acc,
                   <ul className="autosuggest-dropdown-list" key={title}>
                     <li className="autosuggest-dropdown-header">{title}</li>
                     {matches
-                      .slice(0, 100) // Limit DOM rendering to 100 results, as DOM rendering is sloooow.
```
Comment (LeviHarrison) [inline web/ui/react-app/src/pages/graph/ExpressionInput.tsx:97]: Pretty sure this was intentional 🤣

**Review by LeviHarrison — APPROVED** [review]

**Comment by Nexucis** [comment]
I included in this PR the upgrade to typescript v4 since it doesn't bring more changes.

Like that it will only remain the upgrade to react v17 in the PR #9284

**Comment by Nexucis** [comment]
ah never mind it breaks all try catch with the error will revert this commit and do it in another PR

**Review by SuperQ — APPROVED** [review]
Changes look mostly mechanical, didn't see any issues in the deploy preview.

LGTM

**Comment by roidelapluie** [comment]
Thanks!

File: web/ui/react-app/package.json [hunk 3]
```diff
@@ -84,23 +81,18 @@
     "@types/reactstrap": "^8.7.2",
     "@types/sanitize-html": "^1.20.2",
     "@types/sinon": "^9.0.4",
-    "@typescript-eslint/eslint-plugin": "2.x",
-    "@typescript-eslint/parser": "2.x",
```
Comment (juliusv) [inline web/ui/react-app/package.json:88]: Just curious, what happened to these two deps, were we just not using them?

Comment (Nexucis) [reply web/ui/react-app/package.json:88]: it's handled by the react-script. So no need to take care of them

Comment (Nexucis) [reply web/ui/react-app/package.json:88]: like that it reduced the number of potentiel conflict between the different dependencies version

**Pull request merged**